Core/Core/View/Base/FullScreenErrorView/FullScreenErrorView.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  FullScreenErrorView.swift
+//  Course
+//
+//  Created by Shafqat Muneer on 5/14/24.
+//
+
+import SwiftUI
+import Theme
+
+public struct FullScreenErrorView: View {
+    
+    public enum ErrorType {
+        case noInternet
+        case noInternetWithReload
+        case generic
+    }
+    
+    private let errorType: ErrorType
+    private var reloadAction: () -> Void = {}
+    
+    public init(
+        type: ErrorType
+    ) {
+        self.errorType = type
+    }
+    
+    public init(
+        type: ErrorType,
+        reloadAction: @escaping () -> Void
```

Comment:
Could we generalize it for the future and call it just `action`?

Reply:
Yes, it's updated with a more generic name.